### PR TITLE
[joi] Add missing definitions to greater and less functions in DateSchema

### PR DESCRIPTION
**Pull request by janari**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/hapijs/joi/blob/v13.6.0/API.md#dategreaterdate
https://github.com/hapijs/joi/blob/v13.6.0/API.md#datelessdate
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/joi/index.d.ts [hunk 1]
```diff
@@ -909,6 +909,28 @@ export interface BinarySchema extends AnySchema {
...
+
+    /**
+     * Specifies that the value must be less than date.
+     * Notes: 'now' can be passed in lieu of date so as to always compare relatively to the current date,
+     * allowing to explicitly ensure a date is either in the past or in the future.
+     * It can also be a reference to another field.
+     */
+    less(date: Date): this;
+    less(date: number): this;
+    less(date: string): this;
```
Comment (SimonSchick) [inline types/joi/index.d.ts]: Can you please add a specific overload for `'now'`, this make it easier to use.

Comment (SimonSchick) [reply types/joi/index.d.ts]: Alternatively you can just merge all of these overloads together, otherwise only the first one will get doc tags.

**Review by SimonSchick — CHANGES_REQUESTED** [review]
Please wait for https://github.com/DefinitelyTyped/DefinitelyTyped/pull/29840/files to be merged.

File: types/joi/index.d.ts (hunk 1 above)
Comment (janari) [reply types/joi/index.d.ts]: I thought about both of the suggestions, but for the consistency sake kept following the same convention that was used with other methods for that specific definition.

**Review by SimonSchick — APPROVED** [review]
Neat, you could optionally create a type for `'now' | Date | number | string | Reference` but this is fine by me.

**Comment by SimonSchick** [comment]
Try rebasing pls.

**Comment by SimonSchick** [comment]
Please rebase, do not merge with master.

**Comment by janari** [comment]
As this pull request has become a real mess, i will close this.

**Comment by SimonSchick** [comment]
You could've just rebased your initial commit and force pushed, there is no need to create a new PR.

**Comment by janari** [comment]
I tried to rebase, but it introduced some merging issues i had to fix manually with which i do not feel comfortable with.

**Comment by janari** [comment]
What i ended up doing was to hard reset my fork and added my initial changes on top of that. Should be fine now.

**Review by nycdotnet — APPROVED** [review]

**Review by rbuckton — APPROVED** [review]

**Pull request merged**